### ISSUE TITLE
Chapter 3 - Classification - shuffling should happen first?

### ISSUE THREAD

**Issue by mnylen** [ageron/handson-ml#86]
Just a quick question - in the Chapter 3 - Classification we split the data to training and test set, then shuffle the training set to ensure all cross-validation folds are similar without missing digits.

Shouldn't it be shuffle first, then split? As far as I understand - and because the data seems to be sorted from 1s to 9s - if we first split and then shuffle, we end up with the training set containing only 1, ..., 8 instances  and the test set containing only 9s.

Wouldn't we want both sets to have proportionally the same amount of every instances?

**Comment by ageron** [comment]
Hi @mnylen ,
Great question! In general, you are absolutely right, it's better to shuffle first and then split, but in the case of MNIST, it is a standard dataset that has already been split into a training set and a test test, and if you want your results to be comparable to other people's results, it's best to use the same training/test set as everyone else. The first 60,000 instances are the training set (with about 6,000 images of each digit, indeed ordered from 0s to 9s), and the last 10,000 are the test set (with about 1,000 images per digit, and again ordered from 0s to 9s).

Note that TensorFlow's `input_data()` function to load MNIST actually returns a `Datasets` object that gives you easy access to the training set and the test set, already split and shuffled for you.  Note that it uses 5,000 instances from the training set as the validation set.

Hope this helps!

**Comment by ageron** [comment]
Closing this issue, please comment if you need any further explanations.
